src/main/kotlin/org/jetbrains/intellij/IntelliJPlugin.kt

```diff
@@ -701,6 +720,11 @@ open class IntelliJPlugin : Plugin<Project> {
 
             task.dependsOn(buildPluginTask)
             task.dependsOn(verifyPluginTask)
+            task.dependsOn(signPluginTask)
+        }
+
+        project.afterEvaluate {
```

Comment:
Why is `afterEvaluate` required here? it's a bit suspicious.

Reply:
If the sign plugin task is enabled we should deploy the signed artifact. In another case, we should deploy an unsigned artifact. I don't know a better way to do this

Reply:
my question is why it's under `afterEvaluate`. Cannot it be unwrapped?

Reply:
Because the value of the `enabled` property is known only after evaluation as far as I understand

Reply:
I see. I don't think this is the good way to handle that. Tasks state being manipulated after evaluation means that  user's data or some other plugins data might be lost. 

I think `enabled` flag is just wrong. `Task` already have one, duplicating in in an extension doesn't make much sense but brings some tricky questions about evaluation order and data loss.